docs/docs/dev_guide/profiles/cprofile.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: default
+title: cProfile Setup and Justification
+parent: Developer's Guide to AaC
+nav_order: 2
+---
+
+# cProfile Setup and Justification
+
+## cProfile Description
+
+`cProfile`, like `profile` is a deterministic profiler. Unlike the Python-backed `Profile`, `cProfile` is backed by the `lsprof` , and is a `C` extension.
+This profiler is the one that comes recommended for most use-cases as it has reasonable overhead and works really well for some long running programs.
+
+With this in mind `cProfile` is quite efficitient and is able to run quite quickly allowing quick and easy profiling of plugin code.   
+## Getting Started
+
+To get started using the `cProfile` profiler you will need to import the profile into a new or existing plugin. 
+
+If you are using `cProfile` this can be done with an import:
+
+```python
+import cProfile
+```
+
+To enable this profiler you can run the following commands against this class:
+
+```python
+""" This is with using the `cProfile` profiler within a ContextManager Object. """
+
+import cProfile
+import pstats
+from pstats import SortKey
+
+with cProfile.Profile() as pr:
+     with plugin_result(plugin_name, write_design_doc_to_directory) as result:
+          s = io,StringIO() // this is used to capture any IO streams
+          ps = pstats.Stats(pr, stream=s).sort_stats(SortKey.TIME)
+          ps.print_stats(10)
+          return result
+```
+
+```python
+""" This is with using the `cProfile` profiler without the ContextManager Object """
+import cProfile
+import pstats
+from pstats import SortKey
+
+pr = cProfile.Profile()
+pr.enable()
+with plugin_result(plugin_name, write_design_doc_to_directory) as result:
+     s = io.StringIO() // this is used to capture any IO streams
+     sortby = SortKey.CUMULATIVE
+     ps = pstats.Stats(pr, stream=s).sort_stats(sortby)
+     ps.print_stats()
+     return result
+```
+
+So lets break this down a bit:
+When you don't sort your results you can get a lot of noise from some functions and methods that are being called and can really
+disorient and add to the confusion of what is going on with the plugin. The above code was run for the `gen_design_doc_impl.py` file
+and within the results as seen with `ps.print_stats(10)` we are filtering the top ten from the line before its sort.
```

Comment:
I reworded this section, the sorting occurs prior to the print stats. `pstats.Stats(pr, stream=s).sort_stats(sortby)` [sorts the stats.](https://docs.python.org/3/library/profile.html#pstats.Stats.sort_stats)
```suggestion
and within the results as seen with `ps.print_stats(10)` we are filtering the top ten sorted lines.
```